textacy/spacy_utils.py

```diff
@@ -90,28 +90,15 @@ def normalized_str(token):
         raise TypeError(msg)
 
 
-def merge_spans(spans, doc):
+def merge_spans(spans):
     """
-    Merge spans so that each only takes up a single token. Since each Span is
-    a view into the doc, merging one invalidates the others. (This will get fixed!)
-    The temporary solution is to gather the information first, before merging.
-    NB: This modifies `doc` in-place!
+    Merge spans *in-place* within parent doc so that each takes up a single token.
```

Comment:
qq: So let's say I have a span `The quick brown fox ate a delicious white rabbit`, and I want to remove adjectives.  If I understand things, this function allows me to merge the resulting spans into `The fox ate a rabbit.'   But does merge keep the syntactic dependencies of the initial span, or does it reparse?